client/layout/index.jsx

```diff
@@ -144,6 +145,17 @@ Layout = React.createClass( {
 		);
 	},
 
+	renderPreview() {
+		if ( config.isEnabled( 'preview-layout' ) && this.props.section.group === 'sites' ) {
```

Comment:
An interesting idea! I am pondering it. It seems like there are trade-offs here since if we move the condition into `DesignPreview` then in a way it tightly binds that class to that config check. OTOH, it may be a temporary config and it's unlikely we'd use the object anywhere else in Calypso...